types/use-analytics/index.d.ts

```diff
@@ -1,32 +1,33 @@
 // Type definitions for use-analytics 0.0
-// Project: https://github.com/DavidWells/analytics/tree/master/packages/use-analytics
-// Definitions by: Harry Cruse <https://github.com/CruseCtrl>
+// Project: https://github.com/DavidWells/analytics
+// Definitions by: Oliver Alonso <https://github.com/boxgames1>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
-import * as React from 'react';
 import { AnalyticsInstance } from 'analytics';
+import * as React from 'react';
 
-export function useAnalytics(): AnalyticsInstance;
+export type TrackFunction = () => void;
+export type PageFunction = () => void;
+export type IdentifyFunction = () => void;
 
+/**
+ *  React hook useAnalytics() for getting analytics functions
+ *
+ * @returns analytics functions: track, page and identify
+ */
+export function useAnalytics(): AnalyticsInstance;
 export function useTrack(): AnalyticsInstance['track'];
 export function usePage(): AnalyticsInstance['page'];
 export function useIdentify(): AnalyticsInstance['identify'];
 
-export const AnalyticsContext: React.Context<AnalyticsInstance>;
+export function withAnalytics(Component: any): any;
 
-export interface AnalyticsProviderProps {
-    children: React.ReactNode;
+export const AnalyticsContext: React.Context<{
     instance: AnalyticsInstance;
-}
-
-export const AnalyticsProvider: React.ComponentType<AnalyticsProviderProps>;
-
-export const AnalyticsConsumer: typeof AnalyticsContext.Consumer;
-
-export interface WithAnalyticsProps {
-    analytics: AnalyticsInstance;
-}
-
-export function withAnalytics<T extends WithAnalyticsProps>(
-    Component: React.ComponentType<T>,
-): React.ComponentType<Omit<T, 'analytics'>>;
+}>;
+export const AnalyticsProvider: React.Provider<{
+    instance: AnalyticsInstance;
+}>;
+export const AnalyticsConsumer: React.Consumer<{
+    instance: AnalyticsInstance;
+}>;
```

Comment:
This isn't actually the correct type. If you look at [the docs on npm](https://www.npmjs.com/package/use-analytics#analyticsconsumer), the way it's used implies it's of type `React.Consumer<AnalyticsInstance>`